src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -65,6 +87,13 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             }
             catch (Exception exception)
             {
+                if (exception.InnerException != null)
+                {
+                    // TypeConverter throws System.Exception wrapping the FormatException,
+                    // so we capture the inner exception.
+                    exception = ExceptionDispatchInfo.Capture(exception.InnerException).SourceException;
```

Comment:
I'd prefer that this be handled explicitly. We know this code can throw a `FormatException` or a wrapped `FormatException` - taking a dependency on the `InnerException` being null in the first case is brittle